src/flux/store.js

```diff
@@ -414,14 +464,56 @@ class Store extends EventEmitter {
 
   importCustomYAML = (customYAML) => {
     logger.log("importCustomYAML - customYAML", customYAML);
-    this.initFlowChart(customYAML);
+    this.createNewFlow(customYAML);
     this.closeModal();
     this.emit("update-flowchart");
   };
 
-  loadExample = (exampleName) => {
-    const flow = exampleYAML[exampleName];
-    this.initFlowChart(flow);
+  loadFlow = (flowId) => {
+    _store.selectedFlow = flowId;
+    this.emit("update-flowchart");
+  };
+
+  updateFlow = (newFlow) => {
+    _store.flows[_store.selectedFlow].flow = newFlow;
+    this.emit("update-flowchart");
+  };
+
+  createNewFlow = (customYAML) => {
+    let prefixString = "Custom Flow";
+
+    let userFlows = Object.entries(_store.flows)
+      .filter(([id, flow]) => flow.name.startsWith(prefixString))
+      .map(([id, flow]) => flow);
+
+    const flowNumbers = userFlows
+      .map((f) => parseInt(f.name.substring(prefixString.length)) || 0)
+      .sort((a, b) => a - b);
+
+    const largestNumber = flowNumbers[flowNumbers.length - 1] || 0;
+
+    const id = nanoid();
+
+    let flow;
+
+    if (customYAML) {
+      const parsed = parseYAML(customYAML);
+      let canvas;
+      try {
+        canvas = parsed.data.with.board.canvas;
+      } catch (e) {
+        canvas = {};
+      }
+      flow = formatForFlowchart(parsed.data.pods, canvas);
+    } else flow = getInitialFlow();
+
+    _store.flows[id] = {
+      name: `${prefixString} ${largestNumber + 1}`,
+      type: "user-generated",
+      flow,
+    };
```

Comment:
why setting this again? wouldn't we get problems if we now start deleting numbers?

Reply:
If we delete a new flow with the highest number, the next new flow will take on that name. If we delete one from the middle, the logic will continue with incrementing the highest number. Keep in mind this is just to set the default name, user will be able to change name. What would be the problem with deleting?

Reply:
That's what I meant, which should not happen.
If a certain flow is flow1 and now you add or remove a flow and what was flow1 is now flow2 and what was flow2 is now flow3 or something like that, that's really confusing.
If a name does not identify a certain flow but only a position in a list, we can just write some index next to the items. That would be less confusing.